WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShippingCreateLabelsView.swift

```diff
@@ -56,7 +56,12 @@ struct WooShippingCreateLabelsView: View {
                         WooShippingServiceView(viewModel: shippingService)
                             .padding(.horizontal, -16)
                     } else {
-                        WooShippingPackageAndRatePlaceholder()
+                        if let storeOptions = viewModel.storeOptions {
+                            WooShippingPackageAndRatePlaceholder(storeOptions: storeOptions)
+                        }
+                        else {
+                            loadingView
```

Comment:
Having a loading view here feels a little odd to me, especially because the data isn't needed for this placeholder view — it just needs to be passed down to its child views.

What do you think about setting the store options as an environment variable, instead? That way we could initialize this view without the `storeOptions` property necessarily being loaded yet, and once we have those options we can pass them to the views that need them. Or we could use some other approach to set/update the store options after initialization.

This would also allow us to use the fallbacks from `ServiceLocator` immediately, and then update them in place if we get fresh data from the backend.

Reply:
We will still need to do pick a place where we want to trigger the fetching/loading the data. And in theory that loading could take a longer time and user could get to a state in UI that needs this data and we need to show that something will be there.

For example, I do not show any loading indicators and user gets to Add Package screen and Custom tab. I still do not have the store options therefor I do not have the dimension and weight units. Should I then show some indicator that something is loading? Because at that state we do not have all the data needed to show the UI and we cannot just show blank white screen.

If I use environment variable again we need to decide when will we trigger fetching/loading so it would basically be the same just a different implementation.

I can do it in a way that I trigger the initial load when we enter label creation flow and then if some part of the app needs the store options and they are not loaded yet I will show some loading indication there, basically showing the loading at the last possible state to avoid blocking user as little as possible.

I will see if I will do that change here or in a separate PR when I see how big of a change would it be.

Reply:
Updated:

Basically loading store options (account settings) when we enter the label creation screen, but showing loading indicator only if I get to the part of the UI that needs the store options and they are not loaded. That way in most cases users will not even see the loading indicator. 😊

Updated screen recordings in description

Reply:
> For example, I do not show any loading indicators and user gets to Add Package screen and Custom tab. I still do not have the store options therefor I do not have the dimension and weight units. Should I then show some indicator that something is loading?

Why not use the dimension and weight units from `ServiceLocator.shippingSettingsService` by default? We can still trigger fetching/loading the account settings when the shipping label flow begins, and update those units in place if they are different from the default we've used.